core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -413,8 +413,10 @@ class FileMessageSet private[kafka](@volatile var file: File,
   /**
    * Read from the underlying file into the buffer starting at the given position
    */
-  def readInto(buffer: ByteBuffer, relativePosition: Int): ByteBuffer = {
-    channel.read(buffer, relativePosition + this.start)
+  def readInto(buffer: ByteBuffer, relativePosition: Int, throttler: Option[Throttler] = None): ByteBuffer = {
+    val bytesRead = channel.read(buffer, relativePosition + this.start)
+    if (bytesRead > 0)
+      throttler.foreach(_.maybeThrottle(bytesRead))
```

Comment:
I'm not really sure about this one. It seems like there isn't much of a benefit in doing it here and it's not clear to me that FileMessageSet should care about throttling.